editor/src/components/editor/actions/actions.tsx

```diff
@@ -5395,6 +5399,124 @@ export const UPDATE_FNS = {
         }
       }
 
+      return {
+        ...updatedEditor,
+        filesModifiedByAnotherUser: updatedEditor.filesModifiedByAnotherUser.concat(
+          action.fullPath,
+        ),
+      }
+    }
+  },
+  UPDATE_EXPORTS_DETAIL_FROM_COLLABORATION_UPDATE: (
+    action: UpdateExportsDetailFromCollaborationUpdate,
+    editor: EditorModel,
+    serverState: ProjectServerState,
+  ): EditorModel => {
+    // When the current user does own the project...
+    if (serverState.isMyProject === 'yes') {
```

Comment:
We're starting to see this same check littered about the code, and it's making me uneasy because in some cases (like here) this will be incompatible with the concept of shared ownership of a project. I'm wondering if it makes sense to have a function (or series of functions) that provide checks around whether a certain type of edit can be made to the project, which would themselves right now check `serverState.isMyProject`, even if all of those functions are then just performing this exact check right now

Reply:
We could have something similar to `isTransientAction` which gates the applicability of actions.

Reply:
Yeah I think that would make sense because at least then the logic would all be captured in a single place. I think we'll still have some awkwardness to deal with when we have to handle whether the sender of the change was the owner or not, but we can burn that bridge when we come to it